### DISCUSSION TITLE
Use Real Serial Port on PC

### DISCUSSION THREAD

**Discussion by momone50** [keirf/flashfloppy#444] [Q&A] [answered]
HI, 
i'd like to know if you can use real serial port on a PC to Write the firmware. I'd like to avoid buying a USB/TTL converter as
it only seems to replace a real serial.

Anyone tried it? Any feedback?

Thanks

**Comment by randybb** [comment, marked as answer]
You can, just you need to drive it from +/- 12V (that is standard RS232) to TTL levels that UART uses. It is simple, you just need to connect it via a driver that will do it for you, like https://www.ti.com/lit/ds/symlink/max232.pdf

**Reply by momone50** [reply]
I can't use the 5V pin on my DB9 connector?

**Reply by momone50** [reply]
It seems i made a confusion. Tell me if i'm wrong:
There is indeed a 5V available on PIN 9, on Serial Port, but i need it to drive the Gotek.
When you speak of +/- 12V, you mean on RX/TX pin? that's it?

**Reply by randybb** [reply]
What is 5 V pin? For a standard RS232 normal voltage level for logical "0" is -12 V and for logical "1" +12 V. It is nicely described here https://electronics.stackexchange.com/questions/274287/what-is-the-difference-between-rs-232-and-ttl-uart

**Reply by randybb** [reply]
Pin 9 is Ring Indicator. Well, you can have a serial port with +/- 5V, but it is still not compatible with TTL voltage levels.

**Reply by momone50** [reply]
5V is compatible with TTL. No?
I didn't practice Serial for a while and it seems i made confusions between UART and SERIAL.
I stupidly googled "Serial 5V pin" and found things about a 5V on Pin 9....
Concretely, and if i'm not mistaken, you need to Drive the Gotek with a +5V and and Ground Pin, and connect RX/TX on it. That's it? So what pin should i use to provide 5V?

**Comment by omikron88** [comment]
It needs additional hardware that translates voltage levels to 3.3V. Such
as MAX3232.


Dne so 6. 3. 2021 21:54 uživatel Roman Priesol <notifications@github.com>
napsal:

**Reply by momone50** [reply]
Thanks, That's what i just learned from @randybb 
I thought there was a 5v on serial ports...

**Comment by momone50** [comment]
The idea behind  my question was to avoid buying a CP2102 USB/TTL converter as i own a real serail port on my PC, i use for development.
It seems i need, nevertheless, a MAX3232 chip to convert the signal to TTL. 
I decided to buy a CP2102 in the end. Thanks for you answers
